code/modules/crafting/table.dm

```diff
@@ -163,4 +174,12 @@
 		else
 			usr << "<span class ='warning'>Construction failed.</span>"
 		busy = 0
-		interact(usr)
\ No newline at end of file
+		interact(usr)
+
+/obj/structure/table/proc/check_table_space()
+	var/Item_amount = 0
+	for(var/obj/item/I in loc)
+		Item_amount++
+	if(Item_amount <= 20) //is the table crowded?
```

Comment:
20 is magic number
Either make it defined somewhere or even better make it per-recipe so some things need more space to make than others